node/outbox.go

```diff
@@ -100,19 +100,95 @@ func (n *Node) trySend(pid, target string) {
 			return
 		}
 
-		assignItem, err := n.tryGetAssign(item.Id, nodes, itemBin)
-		if err != nil {
-			log.Error("outbox try get assignment failed", "pid", pid, "target", target, "itemId", item.Id, "err", err)
+		if len(nodes) == 0 {
 			return
 		}
 
+		var assignItem goarSchema.BundleItem
+		if n.isSelf(nodes[0]) {
+			assignItem, err = n.tryGetLocalAssign(item.Id, nodes, *item)
+			if err != nil {
+				log.Error("outbox try get local assignment failed", "pid", pid, "target", target, "itemId", item.Id, "err", err)
+				return
+			}
+
+		} else {
+			assignItem, err = n.tryGetAssign(item.Id, nodes, itemBin)
+			if err != nil {
+				log.Error("outbox try get assignment failed", "pid", pid, "target", target, "itemId", item.Id, "err", err)
+				return
+			}
+		}
+
 		if err = n.outboxDB.Commit(pid, target, assignItem); err != nil {
 			log.Error("outbox commit failed", "pid", pid, "target", target, "err", err)
 			return
 		}
 	}
 }
 
+func (n *Node) tryGetLocalAssign(msgId string, nodes []registrySchema.Node, item goarSchema.BundleItem) (assignItem goarSchema.BundleItem, err error) {
+	// make channel for assignment result
+	resultChan := make(chan schema.AssignmentResult, 1)
```

Comment:
The channel buffer size of 1 may lead to performance bottlenecks under heavy load. Consider increasing it (e.g., 1000+) to reduce the risk of blocking when processing results concurrently.